app/src/main/java/com/github/se/travelpouch/ui/home/OnboardingScreen.kt

```diff
@@ -98,9 +120,33 @@ fun OnboardingPageContent(page: OnboardingPage, modifier: Modifier = Modifier) {
         Text(
             page.description,
             fontSize = 16.sp,
-            modifier = Modifier.testTag("OnboardingDescription"))
+            modifier = Modifier.testTag("OnboardingDescription"),
+            textAlign = androidx.compose.ui.text.style.TextAlign.Center)
+      }
+}
+
+// Progress indicator to show the current page in the onboarding process
+@Composable
+fun ProgressIndicator(currentPage: Int, totalPages: Int, modifier: Modifier = Modifier) {
+  Box(
+      contentAlignment = Alignment.Center,
+      modifier = modifier.fillMaxWidth().padding(vertical = 8.dp)) {
+        Row(
+            horizontalArrangement = Arrangement.Center,
+            verticalAlignment = Alignment.CenterVertically) {
+              for (i in 0 until totalPages) {
+                Box(
+                    modifier =
+                        Modifier.padding(horizontal = 4.dp)
+                            .size(if (i == currentPage) 12.dp else 8.dp)
+                            .background(
+                                if (i == currentPage) androidx.compose.ui.graphics.Color.Blue
+                                else androidx.compose.ui.graphics.Color.Gray,
+                                shape = RoundedCornerShape(50)))
+              }
+            }
       }
 }
 
 // Data class to define each onboarding page
-data class OnboardingPage(val title: String, val description: String, val imageResId: Int)
+data class OnboardingPage(val title: String, val description: String, val imageResId: Int?)
```

Comment:
Good modularisation, but more documentation of the data class would be appreciated.